src/jit/codegencommon.cpp

```diff
@@ -7419,26 +7425,88 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
             continue;
         }
 
-        var_types loadType = varDsc->lvaArgType();
-        regNumber argReg   = varDsc->lvArgReg;
-        getEmitter()->emitIns_R_S(ins_Load(loadType), emitTypeSize(loadType), argReg, varNum, 0);
+        {
+            var_types loadType = varDsc->lvaArgType();
+            regNumber argReg   = varDsc->lvArgReg;
+            getEmitter()->emitIns_R_S(ins_Load(loadType), emitTypeSize(loadType), argReg, varNum, 0);
 
 #if FEATURE_VARARG
-        if (compiler->info.compIsVarArgs && varTypeIsFloating(loadType))
+            if (compiler->info.compIsVarArgs && varTypeIsFloating(loadType))
+            {
+                regNumber   intArgReg = compiler->getCallArgIntRegister(argReg);
+                instruction ins       = ins_CopyFloatToInt(loadType, TYP_LONG);
+                inst_RV_RV(ins, argReg, intArgReg, loadType);
+            }
+#endif //  FEATURE_VARARG
+        }
+    }
+
+    // If initReg is one of RBM_CALLEE_TRASH, then it needs to be zero'ed before using.
+    if ((RBM_CALLEE_TRASH & genRegMask(initReg)) != 0)
+    {
+        *pInitRegZeroed = false;
+    }
+
+#else // !defined(UNIX_AMD64_ABI)
+
+    // Use R14 & r15 for passing original values of RDI & RSI to callback because they are re-used below.
```

Comment:
>// Use R14 & r15 for passing original values of RDI & RSI to callback because they are re-used below. [](start = 4, length = 101)

Please add a reason why it has to be R14 and R15 only.  I thought earlier we have used R12/R13.